src/tir/transforms/storage_rewrite.cc

```diff
@@ -583,8 +583,10 @@ class StoragePlanRewriter : public StmtExprMutator {
   };
 
   // Checks whether the storage_scope is especially tagged for a specific memory.
+  // Special memory is all combined into a single allocation.
```

Comment:
I've verified that with the special tagged memory helper change we are getting correct behavior for aligned access in the cases that I mentioned to fail previously.